src/OrleansProviders/Streams/Common/PooledCache/IObjectPool.cs

```diff
@@ -23,6 +23,31 @@ public interface IObjectPool<T>
         /// </summary>
         /// <param name="resource"></param>
         void Free(T resource);
+
+        /// <summary>
+        /// Try to find in-use object in this pool by Id 
+        /// </summary>
+        /// <param name="Id"></param>
+        /// <param name="inUseObject"></param>
+        /// <returns></returns>
+        bool TryFindInUseObject(object Id, out T inUseObject);
```

Comment:
nit: casing of Id.
Also please add code comments for parameters too, especially since these are really cryptic